ui-testing/src/main/java/com/qreal/wmp/uitesting/dia/scene/window/SceneWindowImpl.java

```diff
@@ -0,0 +1,170 @@
+package com.qreal.wmp.uitesting.dia.scene.window;
+
+import com.google.common.base.Predicate;
+import com.qreal.wmp.uitesting.dia.scene.Coordinate;
+import com.qreal.wmp.uitesting.dia.scene.elements.Block;
+import com.qreal.wmp.uitesting.dia.scene.elements.SceneElement;
+import com.qreal.wmp.uitesting.exceptions.ElementNotOnTheSceneException;
+import org.jetbrains.annotations.Contract;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+import static com.codeborne.selenide.Selenide.$;
+
+/**
+ * Describes part of the scene, which is shown on browser.
+ */
+public class SceneWindowImpl implements SceneWindow {
+    
+    private static final Logger logger = LoggerFactory.getLogger(SceneWindowImpl.class);
+    
+    private final WebDriver driver;
+    
+    /** Constructor takes links to current scene and current driver. */
+    private SceneWindowImpl(final WebDriver driver) {
+        this.driver = driver;
+    }
+    
+    /**
+     * Moves element to the requested position.
+     *
+     * @param element element to move
+     * @param dist position to move
+     */
+    @Override
+    public void move(final Block element, final Coordinate dist) throws ElementNotOnTheSceneException {
+        Coordinate src = element.getCoordinateOnScene();
+        focus(src);
+        int sizeHor = Double.valueOf($(By.id("SceneWindowHorSize")).innerHtml()).intValue();
+        int sizeVer = Double.valueOf($(By.id("SceneWindowVerSize")).innerHtml()).intValue();
+    
+        Actions actions = new Actions(driver);
+        actions.clickAndHold(element.getInnerSeleniumElement());
+        if (Math.abs(dist.getXAbsolute() - element.getCoordinateOnScene().getXAbsolute()) < sizeHor / 2
```

Comment:
these conditions are very hard to understand, we better move them into separate functions